lib/phoenix/router/helpers.ex

```diff
@@ -9,20 +9,26 @@ defmodule Phoenix.Router.Helpers do
   @doc """
   Callback invoked by url generated in each helper module.
   """
-  def url(_router, %Conn{private: private}) do
-    private.phoenix_endpoint.url
+  def url(router, %Conn{private: %{phoenix_router_url: %URI{} = uri}}) do
+    url(router, uri)
+  end
+  def url(_router, %Conn{private: %{phoenix_router_url: "" <> url}}) do
```

Comment:
This is equivalent to `is_binary(url)`, right?

Reply:
yeah. I'll change it to is_binary when I rewrite the defs to a cause hough